jetstreamext/README.md

```diff
@@ -0,0 +1,138 @@
+[GoDoc-Url]: https://pkg.go.dev/github.com/synadia-io/orbit.go/jetstreamext
+[GoDoc-Image]: https://pkg.go.dev/badge/github.com/synadia-io/orbit.go/jetstreamext.svg
+
+[![License][License-Image]][License-Url]
+[![Go Reference][GoDoc-Image]][GoDoc-Url]
+[![Build Status][Build-Status-Image]][Build-Status-Url]
+[![Go Report Card][ReportCard-Image]][ReportCard-Url]
+
+JetStream Extensions is a set of utilities providing additional features to `jetstream` pacakge in nats.go client.
+
+## Installation
+
+```bash
+go get github.com/synadia-io/orbit.go/jetstreamext
+```
+
+## Utilities
+
+### GetBatch and GetLastMessagesFor
+
+`GetBatch` and `GetLastMessagesFor` are utilities that allow you to fetch multiple messages from a JetStream stream.
+Responses are returned in an iterator, which you can range over to receive messages.
+
+#### GetBatch
+
+`GetBatch` fetches a `batch` of messages from a provided stream, starting from
+either the lowest matching sequence, from the provided sequence, or from the
+given time. It can be configured to fetch messages from matching subject (which
+may contain wildcards) and up to a maximum byte limit.
+
+Examples:
+
+- fetching 10 messages from the beginning of the stream:
+
+```go
+msgs, err := jetstreamext.GetBatch(ctx, js, "stream", 10)
+if err != nil {
+    // handle error
+}
+for msg, err := range msgs {
+    if err != nil {
+        // handle error
+    }
+    fmt.Println(string(msg.Data))
+}
+```
+
+- fetching 10 messages from the stream starting from sequence 100 and matching subject:
+
+```go
+msgs, err := jetstreamext.GetBatch(ctx, js, "stream", 10, jetstreamext.GetBatchSeq(100), jetstreamext.GetBatchSubject("foo"))
+if err != nil {
+    // handle error
+}
+// process msgs
+```
+
+- fetching 10 messages from the stream starting from time 1 hour ago:
+
+```go
+msgs, err := jetstreamext.GetBatch(ctx, js, "stream", 10, jetstreamext.GetBatchStartTime(time.Now().Add(-time.Hour)))
+if err != nil {
+    // handle error
+}
+// process msgs
+```
+
+- fetching 10 messages or up to provided byte limit:
+
+```go
+msgs, err := jetstreamext.GetBatch(ctx, js, "stream", 10, jetstreamext.GetBatchMaxBytes(1024))
+if err != nil {
+    // handle error
+}
+// process msgs
+```
+
+#### GetLastMessagesFor
+
+`GetLastMessagesFor` fetches the last messages for the specified subjects from the specified stream. It can be optionally configured to fetch messages up to the provided sequence (or time), rather than the latest messages available. It can also be configured to fetch messages up to a provided batch size.
+The provided subjects may contain wildcards, however it is important to note that the `GetLastMessagesFor` utility will match a maximum of 1024 subjects.
+
+Responses are returned in an iterator, which you can range over to receive messages.
+
+Examples:
+
+- fetching last messages from the stream for the provided subjects:
+
+```go
+msgs, err := jetstreamext.GetLastMessagesFor(ctx, js, "stream", []string{"foo", "bar"})
```

Comment:
Is it a deliberate choice to switch to `Message` from `Msg` in Core NATS client?

I'm not against it, just wondering.

Reply:
Not really, I'll change it to `Msgs` for consistency.